src/UserGuide/develop/QuickStart/InterfaceDefinition/InterfaceDefinition-Python.md

```diff
@@ -0,0 +1,277 @@
+    """
+    def write_table(self, tablet: Tablet)
+      
+      
+    """
+    Close TsFileTableWriter and will flush data automatically.
+    :return: no return value.
+    """
+    def close(self)
+
+```
+
+
+
+### Tablet definition
+
+You can use Tablet to insert data into TsFile in batches, and you need to release the space occupied by the Tablet after use.
+
+```Python
+class Tablet(object)
+    """
+    A pre-allocated columnar data container for batch data with type constraints.
+
+    Initializes:
+    - column_name_list: Ordered names for data columns
+    - type_list: TSDataType values specifying allowed types per column
+    - max_row_num: Pre-allocated row capacity (default 1024)
+
+    Creates timestamp buffer and typed data columns, with value range validation ranges
+    for numeric types.
+    """
+
+    def __init__(self, column_name_list: list[str], type_list: list[TSDataType],
+                 max_row_num: int = 1024)
+    
+    
+    def set_table_name(self, table_name: str)
+    def get_column_name_list(self)
+    def get_data_type_list(self)
+    def get_timestamp_list(self)
+    def get_target_name(self)
+    def get_value_list(self)
+    def get_max_row_num(self)
+    def add_column(self, column_name: str, column_type: TSDataType)
+    def remove_column(self, column_name: str)
+    def set_timestamp_list(self, timestamp_list: list[int])
+    def add_timestamp(self, row_index: int, timestamp: int)
+    def add_value_by_name(self, column_name: str, row_index: int, value: Union[int, float, bool, str, bytes])
+    def add_value_by_index(self, col_index: int, row_index: int, value: Union[int, float, bool, str, bytes])
+    def get_value_by_index(self, col_index: int, row_index: int)
+    def get_value_by_name(self, column_name: str, row_index: int)
+    def get_value_list_by_name(self, column_name: str)
+```
+
+
+
+## Read  Interface
+
+### TsFileReader
+
+```python
+class TsFileReader:
+    """
+    Read and query table data from TsFiles.
+    """
+    
+    
+    """
+    Initialize a TsFile reader for the specified file path.
+    """
+    def __init__(self, pathname)
+
+
+    """
+    Execute a time range query on specified table and columns.
+    :return: query result handler.
+    """
+    def query_table(self, table_name : str, column_names : List[str],
+                    start_time : int = 0, end_time : int = 0) -> ResultSetPy
+
+    """
+    Get table's schema with specify table name.
+    """
+    def get_table_schema(self, table_name : str)-> TableSchema
+
+
+    """
+    Get all tables schemas
+    """
+    def get_all_table_schemas(self) ->dict[str, TableSchema]
+
+
+    """
+    Close TsFile Reader, if reader has result sets, invalid them.
+    """
+    def close(self)
+
+    
+    
+```
+
+### ResultSet
+
+
+
+```python
+class ResultSet:
+    """
+    Get data from a query result. When reader run a query, a query handler will return.
+    If reader is closed, result set will not invalid anymore.
```

Comment:
a query result -> a query / a query result set
run -> runs
handler -> result set
will return -> will be returned
will not invalid -> will be invalid